AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/script_canvas_tools_qt.py

```diff
@@ -0,0 +1,448 @@
+import editor_python_test_tools.QtPyO3DEEditor as QtPyO3DEEditor
+import azlmbr.editor as editor
+import azlmbr.math as math
+import azlmbr.bus as bus
+import azlmbr.legacy.general as general
+import azlmbr.scriptcanvas as scriptcanvas
+from consts.general import (SAVE_STRING, NAME_STRING,  WAIT_TIME_3, WAIT_FRAMES_200, ENTITY_STATES)
+from consts.scripting import (ASSET_EDITOR_UI, NODE_PALETTE_UI, NODE_PALETTE_QT, TREE_VIEW_QT, SEARCH_FRAME_QT,
+                              SEARCH_FILTER_QT,  NODE_INSPECTOR_TITLE_KEY, SCRIPT_EVENT_UI, PARAMETERS_QT,
+                              NODE_INSPECTOR_QT, NODE_INSPECTOR_UI, SCRIPT_CANVAS_COMPONENT_PROPERTY_PATH)
+from consts.asset_editor import (EVENTS_QT, DEFAULT_SCRIPT_EVENT, SAVE_ASSET_AS)
+"""
+Editor Qt Object container for easy access
+"""
+qtpy_o3de_editor = QtPyO3DEEditor.QtPyO3DEEditor()
+
+class Tests():
+    new_event_created = ("New Script Event created", "Failed to create a new event")
+    child_event_created = ("Child Event created", "Failed to create Child Event")
+    parameter_created = ("Successfully added parameter", "Failed to add parameter")
+    parameter_removed = ("Successfully removed parameter", "Failed to remove parameter")
+
+def click_menu_option(window, option_text):
+    """
+    function for clicking an option from a Qt menu object. This function bypasses menu groups or categories. for example,
+    if you want to click the Open option from the "File" category provide "Open" as your menu text instead of "File" then "Open".
+
+    param window: the qt window object where the menu option is located
+    param option_text: the label string used in the menu option that you want to click
+
+    returns none
+    """
+    action = pyside_utils.find_child_by_pattern(window, {"text": option_text, "type": QtWidgets.QAction})
+    action.trigger()
+
+def save_script_event_file(self, file_path):
+    """
+    function for saving a script event file with a user defined file path. Requires asset editor qt object to be initialized
+    and any required fields in the asset editor to be filled in before asset can be saved.
+
+    param self: the script object calling this function
+    param file_path: full path to the file as a string
+
+    returns: true if the Save action is successful and the * character disappears from the asset editor label
+    """
+    editor.AssetEditorWidgetRequestsBus(bus.Broadcast, SAVE_ASSET_AS, file_path)
+    action = pyside_utils.find_child_by_pattern(self.asset_editor_menu_bar, {"type": QtWidgets.QAction, "iconText": SAVE_STRING})
+    action.trigger()
+    # wait till file is saved, to validate that check the text of QLabel at the bottom of the AssetEditor,
+    # if there are no unsaved changes we will not have any * in the text
+    label = self.asset_editor.findChild(QtWidgets.QLabel, "textEdit")
+    return helper.wait_for_condition(lambda: "*" not in label.text(), WAIT_TIME_3)
+
+
+def initialize_asset_editor_object(self):
```

Comment:
Refactor to an asset editor uitls file or asset editor QtPy class.

Reply:
made a note to migrate this behavior over to the O3DE and asset editor classes when we reach a test script that requires them